prisma/seeds/prices.ts

```diff
@@ -1,35 +1,127 @@
-import { Prisma } from '@prisma/client'
-export const prices = [
-  { // XECUSD
-    value: new Prisma.Decimal('0.00004095'),
-    createdAt: new Date(),
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 1,
-    quoteId: 1
-  },
-  { // XECCAD
-    value: new Prisma.Decimal('0.00005663'),
-    createdAt: new Date(),
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 1,
-    quoteId: 2
-  },
-  { // BCHUSD
-    value: new Prisma.Decimal('117.9081'),
-    createdAt: new Date(),
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 2,
-    quoteId: 1
-  },
-  { // BCHCAD
-    value: new Prisma.Decimal('163.0735'),
-    createdAt: new Date(),
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 2,
-    quoteId: 2
+import { Price, Prisma } from '@prisma/client'
+import { getAllPricesByTicker } from 'services/priceService'
+import { PRICE_FILE_MAX_RETRIES, TICKERS, NETWORKS, QUOTES, KeyValueAny } from 'constants/index'
+import moment from 'moment'
+
+import * as fs from 'fs'
+import * as path from 'path'
+import { promisify } from 'util'
+
+interface PriceFileData {
+  ticker: string
+  date: string
+  priceInCAD: string
+  priceInUSD: string
+}
+
+export const PATH_PRICE_CSV_FILE = path.join('prisma', 'seeds', 'prices.csv')
+
+async function writeToFile (fileName: string, content: PriceFileData[]): Promise<void> {
+  const writeFile = promisify(fs.writeFile)
+
+  const headers = ['ticker', 'date', 'priceInCAD', 'priceInUSD']
+  const rows = content.map(({ ticker, date, priceInCAD, priceInUSD }) => [ticker, date, priceInCAD, priceInUSD])
+  const csv = [headers, ...rows].map(row => row.join(',')).join('\n')
+
+  await writeFile(fileName, csv, 'utf8')
+}
+
+export async function createPricesFile (): Promise<void> {
+  console.log('Price file creation started')
+  const start = moment()
+  const prices: PriceFileData[] = []
+
+  await Promise.all(Object.values(TICKERS).map(async (ticker) => {
+    const pricesByTicker = await getAllPricesByTicker(ticker)
+
+    pricesByTicker?.forEach(price => {
+      prices.push({
+        ticker,
+        date: price.day,
+        priceInCAD: price.Price_in_CAD,
+        priceInUSD: price.Price_in_USD
+      })
+    })
+  }))
+
+  prices.sort((a, b) => {
+    if (a.ticker < b.ticker) return -1
+    if (a.ticker > b.ticker) return 1
+    if (a.date < b.date) return -1
+    if (a.date > b.date) return 1
+    return 0
+  })
+  await writeToFile(PATH_PRICE_CSV_FILE, prices)
+
+  const finish = moment()
+  console.log(`\n\nstart: ${start.format('HH:mm:ss')}\nfinish: ${finish.format('HH:mm:ss')}\nduration: ${(finish.diff(start) / 1000).toFixed(2)} seconds`)
+}
+
+async function readCsv (filePath: string): Promise<string[][]> {
+  return await new Promise((resolve, reject) => {
```

Comment:
There is, since the `fs.readFile` is not async await (it uses callback), we have to wrap it in a promise to be able to wait on it upon the method `readCsv` call

Reply:
I believe you could use `fs.promises.readFile` as you did with `fs.promises.stat`

Reply:
indeed! done